### PR TITLE
Update CODEOWNERS

### PR DESCRIPTION
**Pull request by obelisk**
Adds foundations as a fallback for Product Security on these changes to help with devspeed.

### CODE REVIEW

**Review by jmank88 — APPROVED** [review]

**Review by patrickhuie19 — APPROVED** [review]

File: .github/CODEOWNERS
Comment (sebawo) [inline .github/CODEOWNERS:121]: @skudasov @patrickhuie19 IMO @smartcontractkit/test-tooling-team needs to be added for /integration-tests go.mod's

File: .github/CODEOWNERS
Comment (sebawo) [inline .github/CODEOWNERS:118]: very often, @smartcontractkit/releng is asked for help with packages and artifacts - IMO, it's worth adding the RE team as a code owner for /contracts, Dockerfiles, and go.mods. etc.
@chainchad, what are your thoughts?

**Review by dhaidashenko — COMMENTED** [review]
Should we also remove `.changeset` dir from the codeowners?
In this [PR](https://github.com/smartcontractkit/chainlink/pull/14635/files) all of the changes are within pkg owned by framework pod, but PR still requires approval from foundations

**Review by jmank88 — APPROVED** [review]

File: .github/CODEOWNERS
Comment (chainchad) [reply .github/CODEOWNERS:118]: Agreed for Dockerfile's. For contracts, at the very least, `@smartcontractkit/releng` should be on:

```
contracts/package.json
package.json
```

As these are modified when incrementing a version for a new release.

Comment (chainchad) [reply .github/CODEOWNERS:118]: Here's what I think we should add:

```
/.goreleaser.*.yaml @smartcontractkit/releng
**/*.Dockerfile @smartcontractkit/releng @smartcontractkit/foundations 
**/Dockerfile @smartcontractkit/releng @smartcontractkit/foundations
**/package.json @smartcontractkit/releng @smartcontractkit/prodsec-public
```

**Pull request closed without merging**
